Tests/SwiftParserTest/translated/EscapedIdentifiersTests.swift

```diff
@@ -99,4 +99,201 @@ final class EscapedIdentifiersTests: ParserTestCase {
+      `escaped reference`(`Escaped Type`())
+      """
+    )
+  }
+
+  func testEscapedIdentifiers12() {
+    assertParse(
+      """
+      func `+ start with operator`() {}
+      func `end with operator +`() {}
+      func ` + `() {}
+      """
+    )
+  }
+
+  func testEscapedIdentifiers13() {
+    assertParse(
+      """
+      func `// not a comment`() {}
+      func `/* also not a comment */`() {}
+      func `func dontDoThis() {}`() {}
+      """
+    )
+  }
+
+  func testEscapedIdentifiers14() {
+    assertParse(
+      """
+      let `@atSign` = 0
+      let `#octothorpe` = 0
+      """
+    )
+  }
+
+  func testEscapedIdentifiers15() {
+    assertParse(
+      """
+      @propertyWrapper
+      struct `@PoorlyNamedWrapper`<`The Value`> {
+        var wrappedValue: `The Value`
+      }
+      struct WithWrappedProperty {
+        @`@PoorlyNamedWrapper` var x: Int
+      }
+      """
+    )
+  }
+
+  func testEscapedIdentifiers16() {
+    assertParse(
+      """
+      let 1️⃣`+` = 0
+      let 2️⃣`^*^` = 0
+      let 3️⃣`.` = 0
+      let 4️⃣`?` = 0
+      func 5️⃣`+`(lhs: Int, rhs: Int) -> Int
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "a raw identifier cannot contain only operator characters"
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "a raw identifier cannot contain only operator characters"
+        ),
+        DiagnosticSpec(
+          locationMarker: "3️⃣",
+          message: "a raw identifier cannot contain only operator characters"
+        ),
+        DiagnosticSpec(
+          locationMarker: "4️⃣",
+          message: "a raw identifier cannot contain only operator characters"
+        ),
+        DiagnosticSpec(
+          locationMarker: "5️⃣",
+          message: "a raw identifier cannot contain only operator characters"
+        ),
+      ]
+    )
+  }
+
+  func testEscapedIdentifiers17() {
+    assertParse(
+      """
+      1️⃣`multiline is
+      not allowed` = 5
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "extraneous code at top level"
+        )
+      ]
+    )
+  }
+
+  func testEscapedIdentifiers18() {
+    assertParse(
+      """
+      `null1️⃣\u{0000}is not allowed` = 5
+      `unprintable ascii2️⃣\u{007f}is not allowed` = 10
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "unprintable ASCII character found in source file"
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "unprintable ASCII character found in source file"
+        ),
+      ]
+    )
+  }
+
+  func testEscapedIdentifiers19() {
+    assertParse(
+      """
+      1️⃣`` = 5
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "a raw identifier cannot be empty"
+        )
+      ]
+    )
+  }
```

Comment:
Do we diagnose raw identifiers that only consist of whitespace? As far as I can tell, they are forbidden by the proposal.

Reply:
We weren't; thanks for catching that. Fixed.